website/docs/language/style.mdx

```diff
@@ -0,0 +1,729 @@
+
+For more information, refer to the [local values documentation](/terraform/language/values/locals) and the [Simplify Terraform configuration with locals](/terraform/tutorials/configuration-language/locals) tutorial. 
+
+## Provider aliasing
+
+Provider aliasing lets you define multiple `provider` blocks for the same Terraform provider. For example, this is useful if you are using a provider that can only target a single region, but you want to deploy to multiple regions. The `provider` meta-argument for resources and the `providers` meta-argument for modules lets you specify which provider to use.
+
+<CodeBlockConfig filename="providers.tf" hideClipboard>
+
+```hcl
+# The default provider configuration; resources that begin with `aws_` will use
+# it as the default, and it can be referenced as `aws`.
+provider "aws" {
+  region = "us-east-1"
+}
+
+# Additional provider configuration for west coast region; resources can
+# reference this as `aws.west`.
+provider "aws" {
+  alias  = "west"
+  region = "us-west-2"
+}
+```
+
+</CodeBlockConfig>
+
+<CodeBlockConfig filename="main.tf" hideClipboard>
+
+```hcl
+resource "aws_instance" "example" {
+  # This "provider" meta-argument selects the AWS provider
+  # configuration whose alias is "west", rather than the
+  # default configuration.
+  provider = aws.west
+
+  # ...
+}
+
+
+module "aws_vpc" {
+  source = "./aws_vpc"
+  providers = {
+    aws = aws.west
+  }
+}
+```
+
+</CodeBlockConfig>
+
+Any provider block that does not define the `alias` parameter is considered the default provider configuration. We recommend that you always include a default provider configuration and that you define all of your providers in the same file. If you define multiple instances of a provider, define the default instance first. For non-default providers, define the `alias` as the first parameter of the `provider` block.
+
+## Dynamic resource count
+
+The `for_each` and `count` meta-arguments let you create multiple resources from a single `resource` block depending on run-time conditions. You can use these meta-arguments to make your code flexible and remove the need to duplicate resource blocks. If your instances are almost identical, `count` is appropriate. If some of their arguments need distinct values that cannot be directly derived from an integer, it is safer to use `for_each`.
+
+The `for_each` meta-argument accepts a `map` or `set` value, and Terraform will create an instance of that resource for each element in the value you provide. In the following example, Terraform creates an `aws_instance` for each of the strings defined in the `web_instances` variable: "ui", "api", "db" and "metrics". The example uses `each.key` to give each instance a unique name. The `web_private_ips` output uses a [for expression](/terraform/language/expressions/for) to create a map of instance names and their private IP addresses, while the `web_ui_public_ip` output addresses the instance with the key "ui" directly.
```

Comment:
do we need this context here? is it enough to just generally say that for_each accepts a map input?